transmission-card.js

```diff
@@ -78,258 +100,279 @@ class TransmissionCard extends HTMLElement {
-           <div class="titleitem"><ha-icon-button icon="mdi:turtle" title="turtle mode" id="turtle"></ha-icon-button></div>
-           <div class="titleitem"><ha-icon-button icon="mdi:stop" title="start/stop all" id="start"></ha-icon-button></div>
-           <div><p id="ttype"></p></div>
-         </div>
+  renderTorrent(torrent) {
+    return html
+    `
+      <div class="progressbar">
+          <div class="${torrent.state} progressin" style="width:${torrent.percent}%">
+          </div>
+          <div class="name">${torrent.name}</div>
+        <div class="percent">${torrent.percent}%</div>
+      </div>
     `;
+  }
 
-    const root = this.shadowRoot;
+  renderTorrentFull(torrent) {
+    return html`
+    <div class="torrent">
+      <div class="torrent_name">${torrent.name}</div>
+      <div class="torrent_state">${torrent.state}</div>
+      <div class="progressbar">
+        <div class="${torrent.state} progressin" style="width:${torrent.percent}%">
+        </div>
+      </div>
+      <div class="torrent_details">${torrent.percent} %</div>
+    </div>
+    `
+  }
 
-    var ttypeElement = root.getElementById('ttype');
-    if ( htype ) {
-      ttypeElement.style.display = "none";
-    } else {
-      ttypeElement.innerHTML = this._ttype;
-      ttypeElement.addEventListener('click', this._toggleType.bind(this));
+  renderTurtleButton() {
+    if (this.config.hide_turtle) {
+      return html``;
     }
 
-    var turtleElement = root.getElementById('turtle');
-    if ( hturtle ) {
-      turtleElement.style.display = "none";
-    } else {
-      turtleElement.addEventListener('click', this._toggleTurtle.bind(this));
-      turtleElement.className = "turtle_" + this.myhass.states['switch.transmission_turtle_mode'].state;
-    }
+    const state = this.hass.states['switch.transmission_turtle_mode'].state;
+    return html`
+      <div class="titleitem">
+        <ha-icon-button 
+          class="turtle_${state}"
+          icon="mdi:turtle" 
+          @click="${this._toggleTurtle}" 
+          title="turtle mode" 
+          id="turtle">
+        </ha-icon-button>
+      </div>
+    `;
+  }
 
-    var playStartElement = root.getElementById('start')
-    if ( hstartstop ) {
-      playStartElement.style.display = "none";
-    } else {
-      playStartElement.addEventListener('click', this._startStop.bind(this));
-      if (this.myhass.states['switch.transmission_switch'].state === "on") {
-        playStartElement.icon = "mdi:stop";
-      } else {
-        playStartElement.icon = "mdi:play";
-      }
-      playStartElement.className = "start_" + this.myhass.states['switch.transmission_switch'].state;
+  renderStartStopButton() {
+    if (this.config.hide_startstop) {
+      return html``;
     }
-  }
 
-  set hass(hass) {
-    const root = this.shadowRoot;
-    //const config = this._config;
-    this.myhass = hass;
+    const state = this.hass.states['switch.transmission_switch'].state;
```

Comment:
The switch should support the non-default sensor names.

Reply:
👍 Good catch. Fixed.